networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_agent.py

```diff
@@ -393,6 +395,10 @@ def sync_inventory(self):
             timestamp = nsxv3_facada.Timestamp(
                 "last_full_synchronization", self.nsxv3, TransportZones, tz, cfg.CONF.AGENT.sync_full_schedule)
 
+            if self.is_first_boot:
+                self.first_boot()
+                self.is_first_boot = False
+
```

Comment:
Doing the cleanup on startup will cause agent to hold all normal operations till the cleanup is done.
I don't think the cleanup of 50k ipsets is going to be so fast that the agent is done before the agent-liveness-check kicks in that kills the agent.

Please reconsider doing the cleanup in the shallow loop and probably also limit the amount of ipsets to delete.